### PR TITLE
[7.16] Drop non-setting deprecations down to warning level

### PR DESCRIPTION
**Pull request by jakelandis**
Complementary to https://github.com/elastic/elasticsearch/pull/79665 this change
will reduce the deprecation logging level to WARN for (most) all non-settings that 
continue to exist in 8.0+.

===

Note - I identified which logs needed to be reduced from the work at https://github.com/elastic/elasticsearch/pull/80167. In that PR I used IntelliJ to refactor ALL deprecations in 8.0+ to log at warning level. One of those commits [`967264e` (#80167)](https://github.com/elastic/elasticsearch/pull/80167/commits/967264ebca0c22ec2989180a295ffcf4e10015f4)  that got squashed in the final merge was only for non-settings. So that commit has all of the non-settings that still exist in 8.0, so I cherry-picked that commit back to 7.16 and only merged files that merged cleanly. So there are likely a couple critical level that could be adjusted since they had additional logic that would require additional manual evaluation. I also had to update the tests since in 7.16 the default assertion is for critical, but in 8.0+ the default assertion is for warning level. There were only a few outliers and have confidence that the list here is accurate and mostly complete.

### CODE REVIEW

**Comment by jakelandis** [comment]
There are now 

189 usages of org.elasticsearch.common.logging.DeprecationLogger#critical
and
82 uses of org.elasticsearch.common.logging.DeprecationLogger#warn

in the 7.16 branch.

File: x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/transform/transforms/TransformConfigTests.java [hunk 1]
```diff
@@ -511,7 +511,7 @@ public void testRewriteForUpdate() throws IOException {
         assertTrue(transformConfigRewritten.getSettings().getDatesAsEpochMillis());
         assertFalse(transformConfigRewritten.getSettings().getAlignCheckpoints());
 
-        assertWarnings(TransformDeprecations.ACTION_MAX_PAGE_SEARCH_SIZE_IS_DEPRECATED);
+        assertWarnings(org.apache.logging.log4j.Level.WARN, TransformDeprecations.ACTION_MAX_PAGE_SEARCH_SIZE_IS_DEPRECATED);
```
Comment (dakrone) [inline x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/transform/transforms/TransformConfigTests.java:514]: Any particular reason these are fully qualified?

**Review by dakrone — APPROVED** [review]
LGTM, I left one question

**Review by pgomulka — APPROVED** [review]
LGTM

File: x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/transform/transforms/TransformConfigTests.java (hunk 1 above)
Comment (jakelandis) [reply x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/transform/transforms/TransformConfigTests.java]: Weird .. Intellij chose to do that.  Fixed in [`0718f87` (#80374)](https://github.com/elastic/elasticsearch/pull/80374/commits/0718f8716aaa8ff7b7075501e7b258887f4d3af3)

**Comment by jakelandis** [comment]
@elasticmachine update branch

**Pull request merged**